### ISSUE TITLE
survfit.matrix fails when newdata is a single row

### ISSUE THREAD

**Issue by MarcusWalz** [therneau/survival#7]
Here is the example in the documentation, the only change is that newdata is a single row:

```R
etime <- with(mgus2, ifelse(pstat==0, futime, ptime))
event <- with(mgus2, ifelse(pstat==0, 2*death, 1))
event <- factor(event, 0:2, labels=c("censor", "pcm", "death"))

cfit1 <- coxph(Surv(etime, event=="pcm") ~ age + sex, mgus2)
cfit2 <- coxph(Surv(etime, event=="death") ~ age + sex, mgus2)

# predicted competing risk curves for a 72 year old with mspike of 1.2
# (median values)
newdata <- expand.grid(sex="M", age=72, mspike=1.2)

AJmat <- matrix(list(), 3,3)
AJmat[1,2] <- list(survfit(cfit1, newdata, std.err=FALSE))
AJmat[1,3] <- list(survfit(cfit2, newdata, std.err=FALSE))
csurv  <- survfit(AJmat, p0 =c(entry=1, PCM=0, death=0))
```

```
Error in 1:dd[2] : NA/NaN argument
```

A very simple solution is:
```
newdata <- bind_rows(newdata, newdata)
```
thanks,

**Comment by therneau** [comment]
I've been out of town on a trip.
You are correct.  This has been fixed in my local copy.  I hope to get it off to CRAN 
sometime this month (but grants are due.)

Terry

**Comment by therneau** [comment]
fixed in version 2.41
